timeseries_to_csv.py

```diff
@@ -55,54 +66,81 @@ def parse_args(args: Sequence[str]) -> argparse.Namespace:
     return parser.parse_args(args)
 
 
-def main():
-    args = parse_args(sys.argv)
-    rc = RestClient(args.host, args.port)
-    rc.login(args.user, args.passwd);
-
+def calculate_begin_time(csv_file_name: str) -> int:
+    """
+    At most, we'll grab 1 day of data, but if we already have some data
+    present, we can just request data since then.
+    """
     last_line = None
-    columns = ['iops.read.rate', 'iops.write.rate',
-               'throughput.read.rate', 'throughput.write.rate',
-               'reclaim.deferred.rate', 'reclaim.snapshot.rate']
-
-    csv_file_name = 'qumulo-timeseries-data.csv'
-
     if os.path.exists(csv_file_name):
         # read to the last line in the file
-        with open(csv_file_name, 'rb') as csvfile:
+        with open(csv_file_name, 'r') as csvfile:
             reader = csv.reader(csvfile)
             for row in reader:
                 last_line = row
 
-    # at most we'll have 1 day of data
-    begin_time = int(time.time()) - 60 * 60 * 24
-    # otherwise, pull only the latest data, if we already have a file
     if last_line is not None:
-        begin_time = int(last_line[0]) + 5
-    results = rc.analytics.time_series_get(begin_time = begin_time)
-    num_results = len(results[0]['values'])
-    print(f'Appending {num_results} results to "{csv_file_name}".')
+        return int(last_line[0]) + 5
+    return int(time.time()) - 60 * 60 * 24
+
+
+def read_time_series_from_cluster(
+    host: str,
+    port: int,
+    begin_time: int
+) -> List[Mapping[str, Any]]:
+    """Communicates with the cluster to grab the analytics in time series format"""
+    rest_client = RestClient(host, port)
+    return rest_client.analytics.time_series_get(begin_time=calculate_begin_time())
```

Comment:
rest_client.login() was dropped

Reply:
This is added later -- not as clean of a patch as I would have liked, though 😢